syntax = "proto3";


service RegistryServerService {
    rpc RegisterReplica (RegisterReplicaRequest) returns (RegisterReplicaResponse) {}
  }

message RegisterReplicaRequest {
    string replica_name = 1;
    string ip = 2;
    int32 port = 3;
  }

message RegisterReplicaResponse {
    bool is_replica_primary = 1;
    string primary_replica_ip = 2;
    int32 primary_replica_port = 3;
  }